tests/src/it/scala/akka/kafka/TransactionsPartitionedSourceSpec.scala

```diff
@@ -33,6 +33,12 @@ class TransactionsPartitionedSourceSpec extends SpecBase
   with Repeated {
 
   val replicationFactor = 2
+  // It's possible to get into a livelock situation where the `restartAfter` interval causes transactions to abort
+  // over and over.  This can happen when there are a few partitions left to process and they can never be fully
+  // processed because we always restart the stream before the transaction can be completed successfully.
+  // The `maxRestarts` provides an upper bound for the maximum number of times we restart the stream so if we get
+  // into a livelock it can eventually be resolved by not restarting any more.
+  val maxRestarts = new AtomicInteger(1000)
```

Comment:
This situation was occurring before and after this fix.  This should make the tests _always_ complete, it just places an upper bound on the number of restarts.